smart_open/tests/test_smart_open.py

```diff
@@ -872,6 +872,18 @@ def write_read_assertion(self, test_file):
 
         if os.path.isfile(test_file):
             os.unlink(test_file)
+    
+    def close_assert(self, test_file):
+        with smart_open.smart_open(test_file, 'wb') as fout: # write after close
+            pass
+        self.assertRaises(ValueError, fout.write, self.TEXT.encode('utf8'))
+        
+        with smart_open.smart_open(test_file, 'rb') as fin: # read after close
+            pass
+        self.assertRaises(ValueError, fin.read)
+        
+        if os.path.isfile(test_file):
+            os.unlink(test_file)
 
```

Comment:
This doesn't test that a filehandle hasn't been closed -- it just tests if fin.read honors that the filehandle isn't closed. You could argue that HttpReadStream should raise ValueError if someone tries to read() after close() has been called, but the fact that it allows it isn't, by itself, a resource leak.